glslc/src/stage_test.cc

```diff
@@ -27,63 +28,40 @@ using shaderc_util::string_piece;
 
 namespace {
 
-TEST(StageDeducer, ValidStages) {
+TEST(DeduceDefaultShaderKindFromFileName, ValidStage) {
   std::stringstream error_stream;
-  EXPECT_EQ(EShLangVertex, glslc::StageDeducer("a.vert")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_vertex_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.vert"));
 
-  EXPECT_EQ(EShLangFragment,
-            glslc::StageDeducer("a.frag")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_fragment_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.frag"));
 
-  EXPECT_EQ(EShLangGeometry,
-            glslc::StageDeducer("a.geom")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_geometry_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.geom"));
 
-  EXPECT_EQ(EShLangTessControl,
-            glslc::StageDeducer("a.tesc")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_tess_control_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.tesc"));
 
-  EXPECT_EQ(EShLangTessEvaluation,
-            glslc::StageDeducer("a.tese")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_tess_evaluation_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.tese"));
 
-  EXPECT_EQ(EShLangCompute,
-            glslc::StageDeducer("a.comp")(&error_stream, "tag"));
-  EXPECT_EQ("", error_stream.str());
+  EXPECT_EQ(shaderc_glsl_default_compute_shader,
+            glslc::DeduceDefaultShaderKindFromFileName("a.comp"));
 }
 
-TEST(StageDeducer, InvalidStages) {
+TEST(DeduceDefaultShaderKindFromFileName, InvalidStage) {
   std::stringstream error_stream;
+  EXPECT_EQ(shaderc_glsl_infer_from_source,
+            glslc::DeduceDefaultShaderKindFromFileName("a.glsl"));
 
-  EXPECT_EQ(EShLangCount, glslc::StageDeducer("a.glsl")(&error_stream, "tag"));
```

Comment:
Tests about StageDeducer are moved to libshaderc's tests. And by porting to libshaderc, we are not using stage deducer any more, so its tests are removed.